src/mainviews/detail-view/job-definition.tsx

```diff
@@ -20,62 +10,97 @@ import {
   FormLabel,
   Stack
 } from '@mui/material';
+
+import cronstrue from 'cronstrue';
+
+import { ButtonBar } from '../../components/button-bar';
 import { ConfirmDialogDeleteButton } from '../../components/confirm-dialog-buttons';
-import { JupyterFrontEnd } from '@jupyterlab/application';
 import {
   ILabeledValueProps,
   LabeledValue
 } from '../../components/labeled-value';
+import { SchedulerService } from '../../handler';
+import { useTranslator } from '../../hooks';
+import { ListJobsTable } from '../list-jobs';
+import {
+  IJobDefinitionModel,
+  JobsView,
+  ICreateJobModel,
+  emptyCreateJobModel
+} from '../../model';
+import { Scheduler as SchedulerTokens } from '../../tokens';
+
+import { timestampLocalize } from './job-detail';
 
 export interface IJobDefinitionProps {
   app: JupyterFrontEnd;
-  model: IJobDefinitionModel;
+  model: IJobDefinitionModel | null;
   refresh: () => void;
   setJobsView: (view: JobsView) => void;
   showJobDetail: (jobId: string) => void;
   showCreateJob: (state: ICreateJobModel) => void;
   editJobDefinition: (jobDefinition: IJobDefinitionModel) => void;
   advancedOptions: React.FunctionComponent<SchedulerTokens.IAdvancedOptionsProps>;
+  reload: () => void;
 }
 
 export function JobDefinition(props: IJobDefinitionProps): JSX.Element {
   const trans = useTranslator('jupyterlab');
   const [displayError, setDisplayError] = useState<string | null>(null);
-
   const ss = useMemo(() => new SchedulerService({}), []);
 
+  const ReloadButton = (
+    <Button variant="contained" onClick={props.reload}>
+      {trans.__('Reload Job Definition')}
+    </Button>
+  );
+
+  const ErrorBanner = displayError && (
+    <Alert severity="error">{displayError}</Alert>
+  );
+
+  if (props.model === null) {
+    return (
+      <>
+        {ErrorBanner}
+        <ButtonBar>{ReloadButton}</ButtonBar>
+      </>
+    );
+  }
+  const model: IJobDefinitionModel = props.model;
```

Comment:
On the lines after this, in some places this new `model` variable is used and in other places `props.model` is used. Is there a special reason to do this?